src/lib/serverVerifyMessage.ts

```diff
@@ -1,27 +1,53 @@
 import "server-only";
 
-import { createPublicClient, http } from 'viem'
+import { createPublicClient, http } from "viem";
 import { mainnet, optimism } from "viem/chains";
-import { type DaoSlug } from "@prisma/client";
+import Tenant from "@/lib/tenant/tenant";
 
-
-export default async function verifyMessage({ address, message, signature, daoSlug }: {
-    address: `0x${string}`,
-    signature: `0x${string}`,
-    message: string,
-    daoSlug: DaoSlug
+export default async function verifyMessage({
+  address,
+  message,
+  signature,
+}: {
+  address: `0x${string}`;
+  signature: `0x${string}`;
+  message: string;
 }) {
-    // Alchemy key
-    const alchemyId = process.env.NEXT_PUBLIC_ALCHEMY_ID!;
+  const { contracts } = Tenant.current();
+
+  const getChain = () => {
+    switch (contracts.token.chainId) {
+      case 1:
+        return mainnet;
+      case 10:
+        return optimism;
+      default:
+        throw new Error(`Invalid chainId: ${contracts.token.chainId}`);
+    }
+  };
+
+  const getRpcUrl = () => {
+    switch (contracts.token.chainId) {
+      case 1:
+        return `https://eth-mainnet.g.alchemy.com/v2/${alchemyId}`;
+      case 10:
+        return `https://opt-mainnet.g.alchemy.com/v2/${alchemyId}`;
+      default:
+        throw new Error(`Invalid chainId: ${contracts.token.chainId}`);
+    }
+  };
+
+  // Alchemy key
+  const alchemyId = process.env.NEXT_PUBLIC_ALCHEMY_ID!;
 
-    const publicClient = createPublicClient({
-        chain: daoSlug === 'OP' ? optimism : mainnet,
-        transport: http(daoSlug === 'OP' ? `https://opt-mainnet.g.alchemy.com/v2/${alchemyId}` : `https://eth-goerli.g.alchemy.com/v2/${alchemyId}`)
-    });
+  const publicClient = createPublicClient({
+    chain: getChain(),
+    transport: http(getRpcUrl()),
+  });
 
-    return await publicClient.verifyMessage({
-        address,
-        message,
-        signature
-    });
-}
\ No newline at end of file
+  return await publicClient.verifyMessage({
+    address,
+    message,
+    signature,
+  });
+}
```

Comment:
I know this logic was there already but we don't need to create a public client, we should have access to provider on the backend already

Reply:
we can use [`ethers.verifyMessage` ](https://docs.ethers.org/v6/api/hashing/#verifyMessage)